internal/configs/virtualserver.go

```diff
@@ -167,7 +169,23 @@ func generateUpstreamStatusMatch(upstreamName string, status string) version2.St
 	}
 }
 
-func generateVirtualServerConfig(virtualServerEx *VirtualServerEx, tlsPemFileName string, baseCfgParams *ConfigParams, isPlus bool) version2.VirtualServerConfig {
+func getEndpointsForVirtualServerEx(namespace string, upstream conf_v1alpha1.Upstream, virtualServerEx *VirtualServerEx, isResolverConfigured bool, isPlus bool) []string {
```

Comment:
name is slightly misleading. the function gets endpoints for an upstream, not a virtualserverEx. Perhaps `getEndpointsForUpstream` or `generateEndpointsForUpstream`?